pexec/managed_process.go

```diff
@@ -472,36 +472,17 @@ func (p *managedProcess) Stop() error {
 		return nil
 	}
 
-	forceKilled, err := p.kill()
+	_, err := p.kill()
 	if err != nil {
 		return err
 	}
 	<-p.managingCh
 
-	if p.lastWaitErr == nil && p.cmd.ProcessState.Success() {
-		return nil
-	}
-
 	if p.lastWaitErr != nil {
-		var unknownStatus bool
-		var errno syscall.Errno
-		if errors.As(p.lastWaitErr, &errno) {
-			// We lost the race to wait before the signal was caught. We're
-			// not going to be able to report any information here about the
-			// process stopping, unfortunately.
-			if errno == syscall.ECHILD {
-				unknownStatus = true
-			}
-		}
-
-		unknownStatus = unknownStatus || isWaitErrUnknown(p.lastWaitErr.Error(), forceKilled)
-		if unknownStatus {
-			p.logger.Debug("unable to check exit status")
-			return nil
-		}
-		return p.lastWaitErr
+		p.logger.Warnw("Stopped module did not exit cleanly", "err", p.lastWaitErr)
```

Comment:
This bit of removing all of the error handling to simply log a warn is the real change. The changes to unix/windows are just removing an unused function. And otherwise changing the expectations of tests.